addons/l10n_ro_efactura/models/ciusro_document.py

```diff
@@ -0,0 +1,174 @@
+            error_messages = [error_element.get('errorMessage') for error_element in error_elements]
+            return {'error': '\n'.join(error_messages)}
+        else:
+            return {'key_loading': root.get('index_incarcare')}
+
+    @api.model
+    def _request_ciusro_fetch_status(self, company, key_loading, session=None):
+        result = self._make_efactura_request(
+            company=company,
+            endpoint='stareMesaj',
+            method='GET',
+            params={'id_incarcare': key_loading},
+            session=session,
+        )
+        if 'error' in result:
+            return result
+
+        root = etree.fromstring(result['content'])
+        error_elements = root.findall('.//ns:Errors', namespaces=NS_STATUS)
+        if error_elements:
+            return {'error': '\n'.join(error_element.get('errorMessage') for error_element in error_elements)}
+
+        state_status = root.get('stare')
+        if state_status in ('nok', 'ok'):
+            return {'key_download': root.get('id_descarcare')}
+        else:
+            return {}
+
+    @api.model
+    def _request_ciusro_download_answer(self, company, key_download, session=None):
+        result = self._make_efactura_request(
+            company=company,
+            endpoint='descarcare',
+            method='GET',
+            params={'id': key_download},
+            session=session,
+        )
+        if 'error' in result:
+            return result
+
+        # E-Factura gives download response in ZIP format
+        zip_ref = zipfile.ZipFile(io.BytesIO(result['content']))
+        signature_file = next(file for file in zip_ref.namelist() if 'semnatura' in file)
+        xml_bytes = zip_ref.open(signature_file)
+        root = etree.parse(xml_bytes)
+        error_element = root.find('.//ns:Error', namespaces=NS_HEADER)
+        if error_element is not None:
+            return {'error': error_element.get('errorMessage')}
+
+        # Pretty-print the XML content of the signature file to be saved as attachment
+        attachment_raw = etree.tostring(root, pretty_print=True, xml_declaration=True, encoding='UTF-8')
+        return {
+            'attachment_raw': attachment_raw,
+            'key_signature': root.find('.//ns:SignatureValue', namespaces=NS_SIGNATURE).text,
+            'key_certificate': root.find('.//ns:X509Certificate', namespaces=NS_SIGNATURE).text,
+        }
+
+    @api.depends('state')
+    def _compute_need_fetch_button(self):
+        for document in self:
+            document.need_fetch_button = document.state == 'invoice_sending'
+
+    def action_l10n_ro_edi_fetch_status(self):
+        """ Fetch the latest response from E-Factura about the XML sent """
+        self.ensure_one()
+        if self.state != 'invoice_sending':
+            raise UserError(_('This document is not currently in sending state'))
+        if not self.key_loading:
+            raise UserError(_('This document does not have a loading key'))
```

Comment:
Is it possible to reach this state? From my understanding, key_loading is not editable

Reply:
it's only possible if the button is not invisible. But well it's impossible for me so I'll remove it.